### ISSUE TITLE
Fix All Traffic pie chart legend and line chart x-axis labels to vertically align

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#2738]
This was previously _attempted_ in #2708, but did not resolve the issue. This is a separate follow-up issue because we didn't have time to make this work in the 1.25.0 release.

Please have a look at #2712: At some point, the legend and the x-axis labels properly aligned. However, that fix introduced a super-large extra whitespace below which made the UI look worse than before. When that margin was removed again, we missed noticing that this once again made the two pieces unaligned 🤦‍♂️ 

Let's find a _proper_ solution here that does not introduce any other problems.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The x-axis labels of the All Traffic line chart should vertically align with the legend of the All Traffic pie chart.
* This must not introduce any other visual regressions. :)

## Implementation Brief

After trying to make the sizes work with set values for way too much time, it seems a set value or a percentage value won't work properly to line up with the pie chart for the Analytics All Traffic widget. I think the best bet is to change the height of the `UserCountGraph` component's `GoogleChart` component based on the window size.

* Use [`useWindowSize`](https://github.com/streamich/react-use/blob/master/docs/useWindowSize.md) to get the size of the window and set the chart height according to the current viewport size.

From my testing something around:

```js
	height: 350,
	width: '100%',
	colors: [ '#1a73e8' ],
	chartArea: {
		left: '1%',
		height: 300,
		top: 18,
		width: '90%',
	},
```

works well for both a desktop and a large desktop viewport, but adjustments need to be made to line up with the legends for both viewports.

* Use the [existing CSS breakpoints](https://github.com/google/site-kit-wp/blob/c61cc0847f47f5c330024a1f4db00803dfa47831/assets/sass/config/_variables.scss#L157-L162) as reference points for when to adjust the `height` vaules of the chart.

### Test Coverage

* No test coverage needed.

### Visual Regression Changes

* Adding VRT tests for the new widget that ensures the legends line up would be good.

## QA Brief

* The legends should vertically align on both smaller and larger desktop screens.

## Changelog entry

* Ensure x-axis labels of the All Traffic line chart are correctly aligned with the pie chart legend.

**Comment by eugene-manuilov** [comment]
> The x-axis labels of the All Traffic line chart should vertically align with the legend of the All Traffic pie chart.

@felixarntz how should this work when we have a multiline legend in the pie chart caused by very long labels? (see #2700) cc @tofumatt

**Comment by felixarntz** [comment]
@eugene-manuilov Good question - I wonder whether then the multiline legend should appear vertically centered compared to the line chart x-axis (i.e. no line is exactly on the same line as the line chart x-axis, but they still look somewhat aligned).

**Comment by felixarntz** [comment]
IB ✅

**Comment by eugene-manuilov** [comment]
> Good question - I wonder whether then the multiline legend should appear vertically centered compared to the line chart x-axis (i.e. no line is exactly on the same line as the line chart x-axis, but they still look somewhat aligned).

@felixarntz it's a hard question. I prefer to see multiline legend top aligned, but it's more of a personal preference. Maybe we should ask Mary about it? :thinking:

**Comment by felixarntz** [comment]
@johnPhillips This is closely intertwined with #2700, which implements a custom version of the legend. Since that issue makes the efforts needed here fairly different, I suggest you hold off of it for now. Let's do 2700 first. In the interest of time, since @asvinb is working on that one, it might be best if he takes over this one here too.

@asvinb Once you updated the PR for #2700 (#2752), can you open a new PR for this issue here, based off your branch from #2700 / #2752? This should allow us to get both of these through tomorrow, and you shouldn't need to wait for @aaemnnosttv or me to approve the first PR before starting the other.

**Comment by aaemnnosttv** [comment]
@asvinb note that @johnPhillips 's approach in his PR #2770 here took a different approach than the IB (using only CSS) and was very close to ready so you might also save a little time by referencing that as a starting point (still based on your branch though as @felixarntz mentioned).

**Comment by asvinb** [comment]
@aaemnnosttv new PR for aligning the legend. Unfortunately we cannot use @johnPhillips approach because we introduced new containers and so the fix does not work.

**Comment by wpdarren** [comment]
### QA Update:  Pass ✅ 

The legends vertically align on both smaller and larger desktop screens. Tested on [Desktop](https://user-images.githubusercontent.com/73545194/107614263-f1043600-6c84-11eb-9fff-44ed929cd4a6.png) / [Phones](https://user-images.githubusercontent.com/73545194/107614268-f497bd00-6c84-11eb-9f14-4d30deec2bdc.png) / [Tablets](https://user-images.githubusercontent.com/73545194/107614280-f95c7100-6c84-11eb-916c-5b3701c81705.png)
